Shell/LineEditor.cpp

```diff
@@ -123,25 +131,33 @@ void LineEditor::tab_complete_first_token(const String& token)
     bool seen_others = false;
     int index = match - m_path.data();
     for (int i = index - 1; i >= 0 && m_path[i].starts_with(token); --i) {
+        suggestions.append(m_path[i]);
         cut_mismatching_chars(completion, m_path[i], token.length());
         seen_others = true;
     }
     for (int i = index + 1; i < m_path.size() && m_path[i].starts_with(token); ++i) {
         cut_mismatching_chars(completion, m_path[i], token.length());
+        suggestions.append(m_path[i]);
         seen_others = true;
     }
+    suggestions.append(m_path[index]);
 
     // If we have characters to add, add them.
     if (completion.length() > token.length())
         insert(completion.substring(token.length(), completion.length() - token.length()));
     // If we have a single match, we add a space, unless we already have one.
     if (!seen_others && (m_cursor == (size_t)m_buffer.size() || m_buffer[(int)m_cursor] != ' '))
         insert(' ');
+
+    return suggestions;
 }
 
-void LineEditor::tab_complete_other_token(String& token)
+// Function returns Vector<String> as assignment is made from return value at callsite
+// (instead of StringView)
+Vector<String> LineEditor::tab_complete_other_token(String& token)
```

Comment:
This one doesn't borrow its strings from `m_path`, so it wouldn't've been possible to returns `StringView`s from it, so no need for the comment here :)

Reply:
Could you give me a quick explanation of the difference of the two? Is `StringView` some kind of reference type?

Reply:
Yes, `String` allocates & keeps its own buffer (on the heap) that it *owns*. `StringView` is a reference into someone else's buffer — usually into someone else's `String`, but could be a static buffer (think `StringView s = "hello friends!";`) or a stack buffer. `StringView` itself is cheap to construct and copy, because it's just a pointer + length, `String` is much heavier. Additionally, `String` guarantees that its buffer is null-terminated, whereas `StringView` doesn't have to be.